### PR TITLE
Fix collation handling for ordering expressions. Fix SqmQuerySpec#setRestriction

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java [hunk 1]
```diff
@@ -933,7 +933,7 @@ private NonSelectQueryPlan buildInsertQueryPlan() {
 		final EntityPersister entityDescriptor = getSessionFactory().getDomainModel().findEntityDescriptor( entityNameToInsert );
 
 		final SqmMultiTableInsertStrategy multiTableStrategy = entityDescriptor.getSqmMultiTableInsertStrategy();
-		if ( multiTableStrategy == null || sqmInsert instanceof SqmInsertValuesStatement ) {
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java:936]: Did you also fix the bug with `insert .... values`?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java:936]: I didn't encounter issues when running tests locally, so I wanted to check how it runs on CI. I'll look into that though if I see errors.

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java:936]: Well yeah, of course, you probably didn't run the documentation tests.

**Comment by gavinking** [comment]
So you said you fixed something. And there's new code there that looks like it's doing something new.

But you have not added any tests.

So what have you fixed that does not require tests?

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java (hunk 1 above)
Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java:936]: I did run the full suite with H2 and it seems H2 also works on CI, but there are errors for other DBs which I will look into tomorrow.

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java:936]: And yes: if you check the build, `HQLTest.hql_insert_example()` is failing.

It's like you think I'm an incompetent idiot who just reports and fixes imaginary bugs.

I just don't understand.

**Comment by beikov** [comment]
What I fixed is, remove the collation from SortSpecification and instead wrap the expression if a collation should be applied for all OrderingExpressions. I didn't test that because it also wasn't tested before, and I first wanted to see if there might be existing tests that fail. I'll add a test though.

**Comment by gavinking** [comment]
>  I didn't test that because it also wasn't tested before

That's not how any of this works.

The most basic principle is: if I fix something, then if possible I add a test for it.

**Comment by gavinking** [comment]
> Add table to readme showing how to test with every database

👏👏👏 Thank you!!

File: documentation/src/test/java/org/hibernate/userguide/hql/HQLTest.java [hunk 2]
```diff
@@ -219,6 +222,15 @@ public void hql_insert_example() {
 				"values (100L, 'Jane Doe')")
 			.executeUpdate();
 
+			//end::hql-insert-example[]
+		});
+	}
+
+	@Test
+	@RequiresDialectFeature(feature = DialectFeatureChecks.SupportsValuesListForInsert.class)
```
Comment (gavinking) [inline documentation/src/test/java/org/hibernate/userguide/hql/HQLTest.java:230]: This is going to conflict with #4574 but I'm just going to let you sort that out.

**Pull request merged**